test/e2e/e2e_test.go

```diff
@@ -53,6 +53,21 @@ var _ = Describe("leaderWorkerSet e2e tests", func() {
 	It("Can deploy lws", func() {
 		lws := testing.BuildLeaderWorkerSet(ns.Name).Obj()
 		Expect(k8sClient.Create(ctx, lws)).To(Succeed())
-		testing.ExpectLeaderWorkerSetAvailable(ctx, k8sClient, lws, "all replicas are ready")
+		testing.ExpectLeaderWorkerSetAvailable(ctx, k8sClient, lws, "All replicas are ready")
+	})
+
+	It("Can rolling update lws", func() {
+		lws := testing.BuildLeaderWorkerSet(ns.Name).Obj()
+		Expect(k8sClient.Create(ctx, lws)).To(Succeed())
+
+		// Wait for leaderWorkerSet ready then update it.
+		testing.ExpectLeaderWorkerSetAvailable(ctx, k8sClient, lws, "All replicas are ready")
+		testing.UpdateWorkerTemplate(ctx, k8sClient, lws)
+
+		// Wait for leaderWorkerSet ready again.
+		testing.ExpectLeaderWorkerSetAvailable(ctx, k8sClient, lws, "All replicas are ready")
```

Comment:
how do we ensure that this is checking the availability status after the update? isn't it possible that we are still checking the previous status before the lws got a chance to pick the update and change the availability status to false?

Reply:
Move this down then we'll check the hash first to make sure we're validating the new revision.

Reply:
I am not sure this will make a difference, it would be nice if we can check that the template hash changed, and then check that the lws is available.

Reply:
I think we already do as you said, so the process is:
- update the lws
- `ExpectValidLeaderStatefulSet` will check that the template hash is already the updated one, see https://github.com/kubernetes-sigs/lws/blob/59f5a761c23609bbc28b9aa0493d6c2b0722a608/test/testutils/validators.go#L121-L124
- `ExpectLeaderWorkerSetAvailable` validate the lws is available.